boxes/generators/cardbox.py

```diff
@@ -0,0 +1,153 @@
+    def __call__(self, length, **kw):
+        depth = self.settings.wallheight-self.thickness*2-10
+        self.edge(length/2-10, tabs=2)
+        self.corner(90)
+        self.edge(depth, tabs=2)
+        self.corner(-180, 10)
+        self.edge(depth, tabs=2)
+        self.corner(90)
+        self.edge(length/2-10, tabs=2)
+
+
+class CardBox(Boxes):
+    """Box for storage of playingcards"""
+    ui_group = "Box"
+
+    def __init__(self):
+        Boxes.__init__(self)
+        self.addSettingsArgs(edges.FingerJointSettings)
+        self.buildArgParser(h=30)
+        self.argparser.add_argument(
+            "--cardwidth",  action="store", type=float, default=65,
+            help="Width of the cards")
+        self.argparser.add_argument(
+            "--cardheight",  action="store", type=float, default=90,
+            help="Height of the cards")
+        self.argparser.add_argument(
+            "--lipwidth",  action="store", type=float, default=10,
+            help="Width of the top lips")
+
+    @property
+    def boxwidth(self):
+        return self.cardwidth*2 + self.thickness*3
+
+    def divider_bottom(self):
+        x = self.boxwidth
+        y = self.cardheight
+
+        self.fingerHolesAt(x/2, 0, y, 90)
+
+    def divider_back_and_front(self):
+        x = self.boxwidth
+        y = self.h - self.thickness
+
+        self.fingerHolesAt(x/2, 0, y, 90)
+
+    def render(self):
+        self.open()
+        h = self.h
+        t = self.thickness
+
+        x = self.boxwidth
+        y = self.cardheight
+
+        s = InsetEdgeSettings(thickness=t)
+        p = InsetEdge(self, s)
+        p.char = "a"
+        self.addPart(p)
+
+        s = FingerHoleEdgeSettings(thickness=t, wallheight=h)
+        p = FingerHoleEdge(self, s)
+        p.char = "A"
+        self.addPart(p)
+
+        self.ctx.save()
+
+        # Lid
+        self.rectangularWall(x, y, "eeee", move="right")
+        # Bottom
+        self.rectangularWall(x, y, "ffff", callback=[self.divider_bottom],
+                             move="right")
+
+        self.ctx.restore()
+        self.rectangularWall(x, y, "EEEE", move="up only")
+        self.ctx.save()
+
+        # Back
+        self.rectangularWall(x, h, "FFEF",
+                             callback=[self.divider_back_and_front],
+                             move="right")
+        # Front
+        self.rectangularWall(x, h, "FFaF",
+                             callback=[self.divider_back_and_front],
+                             move="right")
+
+        self.ctx.restore()
+        self.rectangularWall(x, h, "EEEE", move="up only")
+        self.ctx.save()
+
+        # Outer sides
+        self.rectangularWall(h, y, "fFfF", move="right")
+        self.rectangularWall(h, y, "fFfF", move="right")
+
+        # Inner sides
+        self.rectangularWall(h, y, "eAee", move="right")
+        self.rectangularWall(h, y, "eAee", move="right")
+
+        # Lips
```

Comment:
The lipwidth parameter should go away. Just use t for the width of the lips. The 10mm default basically makes it impossible to use the finger holes at the sides.